crates/bevy_sprite/src/texture_atlas_builder.rs

```diff
@@ -136,21 +136,49 @@ impl TextureAtlasBuilder {
         }
     }
 
-    /// Consumes the builder and returns a result with a new texture atlas.
+    /// Consumes the builder, and returns the newly created texture handle and the assciated atlas layout.
     ///
     /// Internally it copies all rectangles from the textures and copies them
-    /// into a new texture which the texture atlas will use. It is not useful to
+    /// into a new texture atlas will use. It is not useful to
     /// hold a strong handle to the texture afterwards else it will exist twice
     /// in memory.
     ///
+    /// # Usage
+    ///
+    /// ```rust
+    /// # use bevy_sprite::prelude::*;
+    /// # use bevy_ecs::prelude::*;
+    /// # use bevy_asset::*;
+    /// # use bevy_render::prelude::*;
+    ///
+    /// fn my_system(mut commands: Commands, mut textures: ResMut<Assets<Image>>, mut layouts: ResMut<Assets<TextureAtlasLayout>>) {
+    ///     // Declare your builder
+    ///     let mut builder = TextureAtlasBuilder::default();
+    ///     // Customize it
+    ///     // ...
+    ///     // Build your texture and the atlas layout
+    ///     let (atlas_layout, texture) = builder.finish(&mut textures).unwrap();
+    ///     let layout = layouts.add(atlas_layout);
+    ///     // Spawn your sprite
+    ///     commands.spawn(SpriteSheetBundle {
+    ///        texture,
+    ///        atlas: TextureAtlas {
+    ///             layout,
+    ///             index: 0
+    ///         },
+    ///       ..Default::default()
+    ///     });
+    /// }
+    /// ```
+    ///
     /// # Errors
     ///
     /// If there is not enough space in the atlas texture, an error will
     /// be returned. It is then recommended to make a larger sprite sheet.
     pub fn finish(
         self,
         textures: &mut Assets<Image>,
-    ) -> Result<TextureAtlas, TextureAtlasBuilderError> {
+    ) -> Result<(TextureAtlasLayout, Handle<Image>), TextureAtlasBuilderError> {
```

Comment:
I mention it in the changelog of the PR but It might not be enough. @alice-i-cecile What do you think ?

Also I think texture atlas building may become deprecated with the new Assets workflow #8624

Reply:
I'm happy with the notes there.